src/bpfbytecode.cpp

```diff
@@ -0,0 +1,28 @@
+#include "bpfbytecode.h"
+
+#include <stdexcept>
+
+namespace bpftrace {
+
+void BpfBytecode::addSection(const std::string &name,
+                             std::vector<uint8_t> &&data)
+{
+  sections_.emplace(name, data);
+}
+
+bool BpfBytecode::hasSection(const std::string &name) const
+{
+  return sections_.find(name) != sections_.end();
+}
+
+const std::vector<uint8_t> &BpfBytecode::getSection(
+    const std::string &name) const
+{
+  if (!hasSection(name))
+  {
+    throw std::runtime_error("Bytecode is missing section " + name);
+  }
+  return sections_.at(name);
```

Comment:
Not that it matters, but I believe `at()` can also throw. Technically more efficient to use `[]` operator

Reply:
Actually I don't know of the efficient part is true. So ignore that comment I guess

Reply:
Yes, `[]` is marginally more efficient due to not having the bounds checks